interface/slack.py

```diff
@@ -45,14 +44,34 @@ def get_channel_users(self, channel_id):
             "conversation.members",
             channel=channel_id
         )
-        if 'ok' not in response:
+        if 'ok' not in response or response['ok'] is False:
             logging.error("User retrieval "
                           "from channel {} failed with error: {}".
                           format(channel_id, response['error']))
             raise SlackAPIError(response['error'])
         else:
-            json_response = json.loads(response)
-            return json_response["members"]
+            return response["members"]
+
+    def create_channel(self, channel_name):
+        """
+        Create a channel with the given name.
+
+        :return name of newly created channel
+        """
+        logging.debug("Attempting to create channel with name {}".
+                      format(channel_name))
+        response = self.sc.api_call(
```

Comment:
What happens if this function is called and the channel already exists?

Reply:
Response will return with a 'false' in the "ok" field, and a 'name_taken' in the "error" field, I'm pretty sure

Reply:
o wait what do you mean, the entire function or the api_call?

Reply:
Well, both. If I'm understanding correctly, if the name is already taken, the API call will fail and so this will throw an error?

Reply:
Yep

Reply:
Ah, okay. In that case I think this function should handle that case and just return silently - especially since it looks like you don't catch the exception that will be thrown, so I think that would crash the whole server (unless I'm missing something?).

Reply:
Well, return "silently" - it should probably log a warning that the channel already exists.

Reply:
Oh uh yeah if the creation of the bot channel fails, the server dies. How should I handle it? Reattempt to create or just live without one

Reply:
Er well that is one thing to fix, but specifically what I mean here is that if a channel of the same name already exists, this function will throw an exception, which is almost definitely not the behaviour we want - the only time the channel _won't_ already exist is on the very very first launch. So I think in this case the function should notice that the channel already exists, log a warning, and just continue silently.

As to your question (i.e. if it can't create the channel for some _other_ reason), I think both of those would be appropriate - I'm not sure that retrying is that useful, though, because usually it'll be a configuration error of some kind, which won't go away if you just keep retrying 😛 So maybe just log an error and continue without, though keep in mind that'll also make any calls to post to that channel fail as well.

Reply:
Ohhhhh right I get it, the not_taken error should be a good thing not a bad thing